toolchain/sem_ir/typed_insts.h

```diff
@@ -1948,6 +1948,14 @@ struct WitnessType {
   TypeId type_id;
 };
 
+// HasInstCategory is true if T::Kind is an element of AnyInstT::Kinds.
+template <typename AnyInstT, typename T>
+concept HasInstCategory = requires {
```

Comment:
Chandler has asked that we not add concepts as public API, so this should be moved into the `Internal` namespace. I don't really like using `SemIR::Internal` from check, but we already do in a couple of places in import_ref.h, and import_ref is somewhat on the borderline between being part of check and part of semir, so I think it's not terrible.